htcanalyze/time_manager.py

```diff
@@ -0,0 +1,198 @@
+"""Manage times of HTCondor job logs."""
+from datetime import datetime as date_time, timedelta
+from typing import List
+
+
+class TimeManager:
+    """
+    Manage time differences and representation.
+
+    This class creates time differences between the
+    submission, execution and termination date of one job log
+    By that it calculates the runtime.
+
+    Furthermore it can be returned as a dictionary resolving the year,
+    only if the job was running of new year.
+
+    """
+
+    def __init__(self,
+                 submission_date: date_time = None,
+                 execution_date: date_time = None,
+                 termination_date: date_time = None
+                 ):
+        self.submission_date = submission_date
+        self.execution_date = execution_date
+        self.termination_date = termination_date
+        self.waiting_time = timedelta()
+        self.execution_time = timedelta()
+        self.total_runtime = timedelta()
+
+        self.running_over_newyear = False
+
+        self._manage_times()
+
+    def _manage_times(self):
+        """
+        Calculate the waiting, execution and total runtime.
+
+        Check also if the times are overlapping in years.
+
+        """
+        today = date_time.now()
+        today = today.replace(microsecond=0)  # remove microseconds
+
+        # calculate the time difference to last year,
```

Comment:
Check pandas or other modules that deal with time, rather than trying to do it yourself. It's easier, and it's much less likely you'll forget about some weird fringe cases. Time is hard.

Reply:
I would rather avoid that step, as I want to keep imports short and I don
't think, that this is much of a saver. I have created a bunch of tests to make sure all cases are covered. 
To remake this all would not be worth it. But thanks for mentioning, I will consider pandas in future issues with time :)